client/src/components/editor/LinkSourceAnet.js

```diff
@@ -0,0 +1,203 @@
+  overlayColumns: ["Goal", "Author", "Updated"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Report.autocompleteQuery,
+  addon: REPORTS_ICON
+}
+
+const widgetPropsPeople = {
+  objectType: Models.Person,
+  overlayRenderRow: PersonDetailedOverlayRow,
+  overlayColumns: ["Name", "Position", "Location", "Organization"],
+  filterDefs: peopleFilters,
+  queryParams: {},
+  fields: Models.Person.autocompleteQuery,
+  addon: PEOPLE_ICON
+}
+
+const widgetPropsOrganization = {
+  objectType: Models.Organization,
+  overlayRenderRow: OrganizationOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Organization.autocompleteQuery,
+  addon: ORGANIZATIONS_ICON
+}
+
+const widgetPropsPosition = {
+  objectType: Models.Position,
+  overlayRenderRow: PositionOverlayRow,
+  overlayColumns: ["Position", "Organization", "Current Occupant"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Position.autocompleteQuery,
+  addon: POSITIONS_ICON
+}
+
+const widgetPropsLocation = {
+  objectType: Models.Location,
+  overlayRenderRow: LocationOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: { status: Models.Location.STATUS.ACTIVE },
+  fields: Models.Location.autocompleteQuery,
+  addon: LOCATIONS_ICON
+}
+
+const widgetPropsTask = {
+  objectType: Models.Task,
+  overlayRenderRow: TaskSimpleOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: { status: Models.Task.STATUS.ACTIVE },
+  fields: Models.Task.autocompleteQuery,
+  addon: TASKS_ICON
+}
+
+const widgetTypeMapping = {
+  [ENTITY_TYPES.REPORTS]: widgetPropsReport,
+  [ENTITY_TYPES.PEOPLE]: widgetPropsPeople,
+  [ENTITY_TYPES.ORGANIZATIONS]: widgetPropsOrganization,
+  [ENTITY_TYPES.POSITIONS]: widgetPropsPosition,
+  [ENTITY_TYPES.LOCATIONS]: widgetPropsLocation,
+  [ENTITY_TYPES.TASKS]: widgetPropsTask
+}
+
+const LinkSourceAnet = ({ editorState, entityType, onComplete, onClose }) => {
+  const [objectType, setObjectType] = useState(ENTITY_TYPES.REPORTS)
+  const [advancedSelectProps, setAdvancedSelectProps] = useState(
+    widgetTypeMapping[objectType]
+  )
+
+  const advancedSelectRef = useRef()
+
+  function onConfirm(value) {
+    // Retrieve entity URL and label
+    const ModelClass = Models[objectType]
+    const modelInstance = new ModelClass(value)
+    const entityLabel = modelInstance.toString()
+    const entityUrl = ModelClass.pathFor(modelInstance)
+
+    const nextState = createEntity(
+      editorState,
+      entityType.type,
+      {
+        url: entityUrl
+      },
+      entityLabel,
+      "IMMUTABLE"
+    )
+
+    onComplete(nextState)
+  }
+
+  function changeObjectType(newObjectType) {
+    setObjectType(newObjectType)
+    setAdvancedSelectProps(widgetTypeMapping[newObjectType])
+
+    // Filter and type changed, need to update search results
+    advancedSelectRef.current.refreshSearch()
```

Comment:
What's the point of doing this? When you change type the overlay gets closed, when you focus the input it will retrieve new results anyway. But maybe I'm missing something?

Reply:
It used to be stuck after changing the type filter, so a refresh was needed, but I am not observing that behaviour anymore. Removed.

Reply:
I see you didn't remove it, but I tested better and you are right.
Without refreshsearch it doesn't work properly, without the refreshSearch:
- I got to the Anet entity widget in the rich text editor
- I click on People, focus on the input field
- I select All advisors filter type, I see the advisors
- I than click on Organizations, focus on the input field
- I see just as many results as before, and all with empty names
- in the console I see the error that filterDefs is not defined.

So it's ok like this.